src/core/MOM_barotropic.F90

```diff
@@ -4696,6 +4709,17 @@ subroutine barotropic_init(u, v, h, eta, Time, G, GV, US, param_file, diag, CS,
                  "piston velocities.", default=1.0, units="nondim", &
                  do_not_log=.not.CS%linear_wave_drag)
 
+  call get_param(param_file, mdl, "FILTER_ALPHA_M2", am2, &
+                 "Bandwidth parameter of the streaming filter targeting the M2 frequency. "//&
+                 "Must be positive. To turn off filtering, set FILTER_ALPHA_M2 <= 0.0.", &
+                 default=0.0, units="nondim")
+  call get_param(param_file, mdl, "FILTER_ALPHA_K1", ak1, &
+                 "Bandwidth parameter of the streaming filter targeting the K1 frequency. "//&
+                 "Must be positive. To turn off filtering, set FILTER_ALPHA_K1 <= 0.0.", &
+                 default=0.0, units="nondim")
+  om2 = 1.4051890e-4
```

Comment:
The values of `om2` and `ok1` are being set here as hard-coded dimensional constants in [s], but they are used later as though they were being dimensionally rescaled when they are multiplied by dt in [T ~> s].  I believe that this code will fail dimensional consistency testing.

  You should be able to verify that there is in fact a bug here by running the model with various different integer values of the runtime parameter `T_RESCALE_POWER` (try 0, -29 and 29) and you will find that you get different values of these diagnostics with the code that is currently in the PR. The answers will become bitwise identical for these settings when the inconsistencies are corrected.

Instead of hard-coding `om2` and `ok1` as constants, they should be set via `get_param()` calls that would include `..., scale=US%T_to_s, units="s-1")` to convert them into units of [T-1 ~> s-1], and then adjust the units of the `om` argument to `Filt_register()` to [T-1 ~> s-1].

Reply:
The dimensional inconsistency of `om2` and `ok1` has been fixed.